argilla/src/argilla/records/_io/_datasets.py

```diff
@@ -58,6 +59,8 @@ def _record_dicts_from_datasets(dataset: HFDataset) -> List[Dict[str, Union[str,
             Generator[Dict[str, Union[str, float, int, list]], None, None]: A generator of dictionaries to be passed to DatasetRecords.add or DatasetRecords.update.
         """
         record_dicts = []
-        for example in dataset.to_iterable_dataset():
+        if not isinstance(dataset, IterableDataset):
```

Comment:
I think this should also be covered by a try-except statement in case conversion fails, I think we should not do it and just continue without.